### PR TITLE
Use Addressable instead of URI in client.rb

### LINKED ISSUE #6: Exception occured: URI::InvalidURIError - bad URI(is not URI?)
So, let's assume that I perform a GET for the url: ``http://blah.com/api/v1/series%3fmatch%5b%5d%3dup%26match%5b%5d%3dprocess_start_time_seconds%7bjob%3d%27prometheus%27%7d``

The way the route is set up, the
path parameter returns 
``"series?match[]=up&match[]=process_start_time_seconds{job='prometheus'}"``

My index method contains this:

```ruby
 reverse_proxy "http://localhost:9090/api/v1/", path: params[:path], headers: {'Authorization'=> "Basic YWRtaW46YWRtaW4="} do |config|
      config.on_missing do |code, response|
        redirect_to('/') && return
      end
```
I get the error
``Exception occured: URI::InvalidURIError - bad URI(is not URI?): http://localhost:9090/api/v1/series?match[]=up&match[]=process_start_time_seconds{job='prometheus'}``

I'm fairly certain it's because of all the invalid url characters in the query.

### PR DESCRIPTION
**Pull request by bapirex**
This should resolve #6

### CODE REVIEW

File: lib/reverse_proxy/client.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ def request(env, options = {}, &block)
       source_request = Rack::Request.new(env)
 
       # We can pass in a custom path
-      uri = URI.parse("#{url}#{options[:path] || env['ORIGINAL_FULLPATH']}")
+      uri = Addressable::URI.parse("#{url}#{options[:path] || env['ORIGINAL_FULLPATH']}")
```
Comment (axsuul) [inline lib/reverse_proxy/client.rb]: We should have a 

```ruby
require 'addressable/uri`
```

at the top

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-0.5.1
\ No newline at end of file
+0.5.2
```
Comment (axsuul) [inline VERSION]: Let's not modify the version, will handle that on my end 👍

Comment (bapirex) [inline VERSION]: Got it :)

File: lib/reverse_proxy/client.rb (hunk 1 above)
Comment (bapirex) [inline lib/reverse_proxy/client.rb]: Do you want me to submit a new PR?

**Comment by axsuul** [comment]
Thanks so much for your contribution! I've pushed a new release **0.6.0** that contains these changes :)

I've also added you to the contributors list, yay!

**Comment by bapirex** [comment]
Thanks for the credit and this useful gem :)

**Pull request merged**